Packages/com.unity.inputsystem/InputSystem/Editor/AssetEditor/InputActionTreeView.cs

```diff
@@ -0,0 +1,1486 @@
+                buttonRect.width = EditorGUIUtility.singleLineHeight + EditorGUIUtility.standardVerticalSpacing;
+                if (GUI.Button(buttonRect, s_PlusBindingIcon, GUIStyle.none))
+                {
+                    var menu = new GenericMenu();
+                    BuildMenuToAddBindings(menu, actionItem);
+                    menu.ShowAsContext();
+                }
+            }
+        }
+
+        protected override float GetCustomRowHeight(int row, TreeViewItem item)
+        {
+            return 18;
+        }
+
+        protected override Rect GetRenameRect(Rect rowRect, int row, TreeViewItem item)
+        {
+            var textRect = GetTextRect(rowRect, item);
+            textRect.x += 2;
+            textRect.height -= 2;
+            return textRect;
+        }
+
+        private static Rect GetTextRect(Rect rowRect, TreeViewItem item)
+        {
+            var indent = (item.depth + 1) * kColorTagWidth + kFoldoutWidth;
+            var textRect = rowRect;
+            textRect.x += indent;
+            return textRect;
+        }
+
+        #endregion
+
+        // Undo is a problem. When an undo or redo is performed, the SerializedObject may change behind
+        // our backs which means that the information shown in the tree may be outdated now.
+        //
+        // We do have the Undo.undoRedoPerformed global callback but because PropertyDrawers
+        // have no observable life cycle, we cannot always easily hook into the callback and force a reload
+        // of the tree. Also, while returning false from PropertyDrawer.CanCacheInspectorGUI() might one make suspect that
+        // a PropertyDrawer would automatically be thrown away and recreated if the SerializedObject
+        // is modified by undo, that does not happen in practice.
+        //
+        // We could just Reload() the tree all the time but TreeView.Reload() itself forces a repaint and
+        // this will thus easily lead to infinite repaints.
+        //
+        // So, what we do is make use of the built-in dirty count we can get Unity objects. If the count
+        // changes and it wasn't caused by us, we reload the tree. Means we still reload unnecessarily if
+        // some other property on the component changes but at least we don't reload all the time.
+        //
+        // A positive side-effect is that we will catch *any* change to the SerializedObject, not just
+        // undo/redo and we can do so without having to hook into Undo.undoRedoPerformed anywhere.
+
+        private void OnSerializedObjectModified()
+        {
+            serializedObject.ApplyModifiedProperties();
+            UpdateSerializedObjectDirtyCount();
+            Reload();
+            onSerializedObjectModified?.Invoke();
+        }
+
+        private void UpdateSerializedObjectDirtyCount()
+        {
+            m_SerializedObjectDirtyCount = EditorUtility.GetDirtyCount(serializedObject.targetObject);
```

Comment:
Looks like a compile issue here.  no definition for GetDirtyCount

Reply:
Crap...

Good find. Didn't realize this was newly introduced. What if we just take the plunge at this point and with the next package, move on to 2019.1?

Reply:
Sounds good to me, I would LOVE to stop supporting 2018.3!